cmake/AddEventLibrary.cmake

```diff
@@ -87,16 +92,44 @@ macro(add_event_library LIB_NAME)
             set_event_shared_lib_flags("${LIB_NAME}" "${EVENT_SHARED_FLAGS}")
         endif()
 
-        set_target_properties("${LIB_NAME}_shared" PROPERTIES
-            OUTPUT_NAME "${LIB_NAME}"
-            CLEAN_DIRECT_OUTPUT 1)
-        set_target_properties(
-            "${LIB_NAME}_shared" PROPERTIES
-            PUBLIC_HEADER "${HDR_PUBLIC}")
+        if (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+            MATH(EXPR COMPATIBILITY_VERSION "${EVENT_ABI_LIBVERSION_CURRENT}+1")
+            MATH(EXPR CURRENT_MINUS_AGE "${EVENT_ABI_LIBVERSION_CURRENT}-${EVENT_ABI_LIBVERSION_AGE}")
+            set_target_properties(
+                "${LIB_NAME}_shared" PROPERTIES
+                OUTPUT_NAME "${LIB_NAME}-${EVENT_PACKAGE_RELEASE}.${CURRENT_MINUS_AGE}"
+                INSTALL_NAME_DIR "${CMAKE_INSTALL_PREFIX}/lib"
+                LINK_FLAGS "-compatibility_version ${COMPATIBILITY_VERSION} -current_version ${COMPATIBILITY_VERSION}.${EVENT_ABI_LIBVERSION_REVISION}")
+        elseif (SOLARIS OR ${CMAKE_SYSTEM_NAME} MATCHES "Linux")
```

Comment:
Similarly, here, I think what was requested was just `if (NOT WIN32)`. But it seems you have even better handling for macOS now, so maybe what makes sense is:

```
if (WIN32)
    ... simple ...
elseif (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
    ... complicated
else()
    ... the normal standard ELF logic for Linux, FreeBSD, OpenBSD, Solaris, any other unix
endif()
```